### PR TITLE
124 get datawolf person id and use it as the creator

### LINKED ISSUE #124: Get datawolf person ID and use it as the creator ID
So now we all automatically have datawolf person IDs. You can see them all by going to [incore-dev.ncsa.illinois.edu/datawolf/persons](http://incore-dev.ncsa.illinois.edu/datawolf/persons)

So for the playbook, we can pass the userid to datawolf to get their personID: e.g.
`https://incore-dev.ncsa.illinois.edu/datawolf/persons?email=cwang138`

```
[
  {
    "id": "83eb366f-1395-4517-8e86-af0beb82e9f1",
    "deleted": false,
    "firstName": "Chen",
    "lastName": "Wang",
    "email": "cwang138"
  }
]
```

Write a util function that get datawolf user id (maybe in **utils/common.js**) back then use this ID as the `creator ID` in 
**submitExecution()** in **utils/datawolf.js**
```
let requestBody = {
		workflowId: datawolfWorkflowConfig["wfId"],
		creatorId: datawolfWorkflowConfig["creatorId"],
		title: title,
		description: description,
		parameters: parameters,

		datasets: datasets
	};
```
instead of using the default `datawolfWorkflowConfig["creatorId"]` use this retrieved user id

### PR DESCRIPTION
**Pull request by Rashmil-1999**
In this PR, before submitting the execution request to datawolf, first checks if there is a creator id stored with the currently logged-in user, if there is then use that value, otherwise fetch the current user's id and set it as the creator id.

### CODE REVIEW

**Review by longshuicy — COMMENTED** [review]
Could you merge the latest main into your branch so the diff is more concise?
Then maybe start package json at 0.4.7 because I just published 0.4.6

File: src/globals.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ globals["dataServiceBase"] = `${hostname}/data/api/`;
 globals["spaceServiceBase"] = `${hostname}/space/api/`;
 globals["maestroServiceBase"] = `${hostname}/maestro/galveston/`;
 globals["setGravatarURL"] = "https://en.gravatar.com/support/activating-your-account/";
+globals["datawolfPersonsDetail"] = `${hostname}/datawolf/persons?email=`;
```
Comment (longshuicy) [inline src/globals.js]: Could you move this to the datawolf.config.js?

File: src/globals.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ globals["dataServiceBase"] = `${hostname}/data/api/`;
...
 globals["allowUpload"] = true;
 globals["testbedSpace"] = "commresilienceslc";
 globals["testbedGroup"] = "incore_slc_user"; // incore_galveston_user, incore_joplin_user, incore_slc_user
```
Comment (longshuicy) [inline src/globals.js]: could we use incore_joplin_user now since the joplin has the latest workflow

File: src/utils/common.js [hunk 3]
```diff
@@ -51,8 +51,32 @@ export function getCurrUsername(hostname="") {
 	}
 }
 
+// get current users id from datawolf
+export async function getCurrUserDatawolfId(email) {
+	let cookie = cookies.get("DatawolfUserDetails");
+	if (email === "" || undefined || null) {
+		return datawolfWorkflowConfig["creatorId"];
```
Comment (longshuicy) [inline src/utils/common.js]: `datawolfWorkflowConfig` is the default and can only be access in this core repo;
the child repo is actually setting this using redux dispatch. 

I would do 
```
import { datawolfConfig as defaultDatawolfConfig } from "../datawolf.config.js";
```

then 
```
getCurrUserDatawolfId(email, datawolfConfig=defaultDatawolfConfig)
```

Whenever you call this method you can now pass in the datawolfConfig as parameter

File: src/globals.js (hunk 1 above)
Comment (Rashmil-1999) [reply src/globals.js]: Ohh, I thought we might want to keep this here just incase if we need this anywhere else too.

File: src/globals.js (hunk 2 above)
Comment (Rashmil-1999) [reply src/globals.js]: Okay. I will do that

File: src/utils/common.js (hunk 3 above)
Comment (Rashmil-1999) [reply src/utils/common.js]: Oh, I see why you did this everywhere else. I'll update it to this. Thanks

File: src/globals.js (hunk 1 above)
Comment (Rashmil-1999) [reply src/globals.js]: I moved this to the config file but not all the configs in the slc and Galveston might have this entry. What do you think?

File: src/components/childComponents/outputWidgets/HHS.jsx [hunk 4]
```diff
@@ -111,8 +104,6 @@ export default function HHS(props) {
 				</Box>
 			</DialogContent>
 			<DialogActions>
-				<IncoreButtonText onClick={()=>{downloadImg(printRef.current, "HHS");}}
-								  startIcon={<CameraAltIcon/>}>Image</IncoreButtonText>
```
Comment (longshuicy) [inline src/components/childComponents/outputWidgets/HHS.jsx]: Those are actually needed?

File: src/globals.js (hunk 1 above)
Comment (longshuicy) [reply src/globals.js]: we'll match those in the child repos

File: src/datawolf.config.js [hunk 5]
```diff
@@ -5,6 +5,7 @@ const datawolfWorkflowConfig = {
 	wfTitle: "joplin-community-wf-v1",
 	wfId: "05e5ac18-c202-4a8b-973d-01afcae20b14",
 	creatorId: "901e85ca-1a52-4917-83f8-e387b95a259b", // commresileince
+	datawolfPersonsDetail: "https://incore-dev.ncsa.illinois.edu/datawolf/persons?email=",
```
Comment (longshuicy) [inline src/datawolf.config.js]: oh i meant for you to add it in this part of the file
```
let datawolfConfig = {};
let hostname = process.env.INCORE_REMOTE_HOSTNAME || "";

// use default
datawolfConfig["executionGetURL"] = `${hostname}/datawolf/workflows/${datawolfWorkflowConfig["wfId"]}/executions`;
datawolfConfig["executionPostURL"] = `${hostname}/datawolf/executions`;
datawolfConfig["datasetBaseURL"] = `${hostname}/datawolf/datasets`;
datawolfConfig["retreiveTimeInterval"] = 10 * 1000;

export { datawolfWorkflowConfig, datawolfConfig };
```

**Pull request merged**